### DISCUSSION TITLE
Is there a way to import the data from a Greasemonkey 3.14 db into ViolentMonkey?

### DISCUSSION THREAD

**Discussion by TimidScript** [violentmonkey/violentmonkey#1682] [Q&A]
Is there a way to import a script data from a Greasemonkey 3.14 db into ViolentMonkey?

**Comment by tophf** [comment]
Not directly.

1. Find a tool (a viewer/editor for sqlite I guess) and export the data to individual files
2. rename each file as `name.user.js` (there are apps for bulk renaming)
3. pack into a zip, then import it in Violentmonkey

**Comment by TimidScript** [comment]
That does not work. I am trying to import the data and not the scripts. I found a long way, and that's to insert and import and export functionality in the scripts. and manually go through each script. It's not that many, about a dozen. I would have preferred a simpler approach.

**Reply by tophf** [reply]
I don't know how data is stored, but the approach is similar: in addition to name.user.js file you can create `name.options.json` and `name.storage.json` for each script. These files use Tampermonkey format so you can see an example by exporting anything with data from Tampermonkey.

**Reply by TimidScript** [reply]
I exported the database using SQLiteDatabaseBrowser and the structure is different. It's a list of name. value. It was able to import the script fine bur not the data (name.storage.json) file.

**Reply by tophf** [reply]
Indeed, you'll have to convert the structure.
